discover/discover.ml

```diff
@@ -0,0 +1,25 @@
+module C = Configurator.V1
+
+
+let process_output_of command =
+  let chan = Unix.open_process_in command in
+  let res = ref ([] : string list) in
+  let rec process_another_line () =
+    let e = input_line chan in
+    res := e::!res;
+    process_another_line() in
+  try process_another_line ()
+  with End_of_file ->
+    let code = Unix.close_process_in chan in (List.rev !res, code)
+
+let output_text command =
+  let (l, _) = process_output_of command in String.trim (String.concat "\n" l)
+
+
+let output = output_text "uname -s"
```

Comment:
Don't do this as you will break cross compilation. Use `ocaml_config_var_exn` for such environment tests. See my example here: https://github.com/savonet/ocaml-ssl/pull/39/files#diff-81f275ec1ff7428a19d7cd928fb69c94

Also, use the configurator entry point function: `C.main`.